sherpa/astro/tests/test_astro_data.py

```diff
@@ -82,42 +74,42 @@ class test_filter_energy_grid(SherpaTestCase):
         2.58419991, 2.71560001, 2.86159992, 3.08060002, 3.38720012,
         3.5624001, 3.796, 4.02960014, 4.24860001, 4.71579981,
         5.0223999, 5.37279987, 5.89839983, 6.57000017, 9.8696003,
-        14.95040035], float)
+        14.95040035])
 
-    def setUp(self):
-        self.old_level = logger.getEffectiveLevel()
-        logger.setLevel(logging.ERROR)
-        self.pha = DataPHA('', np.arange(46, dtype=float) + 1.,
-                           np.zeros(46),
-                           bin_lo=self._emin,
-                           bin_hi=self._emax)
-        self.pha.units = "energy"
+    pha = DataPHA('', np.arange(46, dtype=float) + 1.,
+                  np.zeros(46),
+                  bin_lo=emin,
+                  bin_hi=emax)
+    pha.units = "energy"
+    return pha
 
-    def tearDown(self):
-        logger.setLevel(self.old_level)
 
-    def test_notice(self):
-        self.pha.notice()
-        self.pha.notice(0.0, 6.0)
-        assert (self._notice == np.asarray(self.pha.mask)).all()
+def test_filter_energy_grid_notice(setUp1):
+    pha = setUp1
+    pha.notice()
+    pha.notice(0.0, 6.0)
 
-    def test_ignore(self):
-        self.pha.notice()
-        self.pha.ignore(0.0, 1.0)
-        self.pha.ignore(3.0, 15.0)
-        assert (self._ignore == np.asarray(self.pha.mask)).all()
+    # Use approx to make it easy to check an array
+    expected = np.ones(46, dtype=bool)
+    expected[44:46] = False
+    assert pha.mask == pytest.approx(expected)
```

Comment:
Sp when working on bools (0 or 1) then the default approx settings are sufficient to distinguish between 0 and 1, so we don't lose anything (in terms of checking). The advantage compared to np.all(...) is that we can get a report of what elements are different, rather than a "something has changed, but I'm not going to tell you what". This has been useful to me in a different PR I'm working on...

The alternatives are to manually print out the differences when you need them (do-able, and at some point once the test works do we need the ease-of-debugging) or to change to something like loop over each element (which you can find in some of our tests), but that's more involved and then you need to also check on the expected length.

Reply:
One place these more specific errors can be useful is in CI tests where we don't always have the same setup (OTS or OS).

I am not planning on making these changes as I don't see a significant improvement and I do see a cost in doing them.